docs/reference/vpc-association-policy.md

```diff
@@ -0,0 +1,82 @@
+| `targetRef`        | *[PolicyTargetReference](https://gateway-api.sigs.k8s.io/geps/gep-713/#policy-targetref-api)* | Yes	     | TargetRef points to the kubernetes `Gateway` resource that will have this policy attached. This field is following the guidelines of Kubernetes Gateway API policy attachment.                                                                                                                                    |
+| `associateWithVpc` | *bool*	                                                                                       | No       | Indicates whether the targetRef Gateway is associated with the current k8s cluster VPC. By default, the Gateway API controller sets this to true if it's not defined in VpcAssociationPolicy.                                                                                                                     |
+| `securityGroupIds` | *string[]*	                                                                                   | No       | Defines security groups applied to the gateway (ServiceNetworkVpcAssociation), it controls the inbound traffic from current cluster workloads to the gateway listeners. Please check the [VPC lattice doc](https://docs.aws.amazon.com/vpc-lattice/latest/ug/security-groups.html) for more detail of this field. |
+
+
+Recommended security group inbound rules:
+
+| Source                                                  | Protocol                                            | Port Range                                      | Comment                                                   |
+|---------------------------------------------------------|-----------------------------------------------------|-------------------------------------------------|-----------------------------------------------------------|
+| Kubernetes cluster VPC CIDR or security group reference | Protocols defined in the gateway's listener section | Ports defined in the gateway's listener section | Allow inbound traffic from current cluster vpc to gateway |
+
+
+### Limitations and Considerations
+
+When attaching a VpcAssociationPolicy to a resource, the following restrictions apply:
+
+* Policies must be attached to *Gateway* resource.
+* The attached resource must exist in the same namespace as the policy resource.
+
+The security Group will not take effect if:
+
+* The targetRef `Gateway` does not exist
+* AssociateWithVpc field set to false
+
+
+**WARNING**
+
+The VPC Lattice `UpdateServiceNetworkVpcAssociation` API cannot be used to remove all security groups.
+That means, if you have a VpcAssociationPolicy attached to a gateway that already applied security groups, update the VPCAssociationPolicy with empty security group ids or delete the whole VPCAssociationPolicy will NOT remove the security groups from this gateway.
```

Comment:
"That means" is redundant. Can replace sentence with "If you have a VpcAssociationPolicy attached to a gateway that already has security groups applied, updating the VpcAssociationPolicy with empty security group ids or deleting the VpcAssociationPolicy will NOT remove the security groups from the gateway."